pycalphad/plot/eqplot.py

```diff
@@ -55,62 +66,116 @@ def eqplot(eq, ax=None, x=None, y=None, z=None, phases=None, **kwargs):
                          if (key == 'T') or (key == 'P') or (key.startswith('X_'))])
-    indep_comp = [key for key, value in conds.items() if isinstance(key, v.Composition) and len(value) > 1]
-    indep_pot = [key for key, value in conds.items() if ((key == v.T) or (key == v.P)) and len(value) > 1]
-    if (len(indep_comp) != 1) or (len(indep_pot) != 1):
-        raise ValueError('Plot currently requires exactly one composition and one potential coordinate')
-    indep_comp = indep_comp[0]
-    indep_pot = indep_pot[0]
-    x = indep_comp if x is None else x
-    y = indep_pot if y is None else y
+    indep_comps = [key for key, value in conds.items() if isinstance(key, v.Composition) and len(value) > 1]
+    indep_pots = [key for key, value in conds.items() if ((key == v.T) or (key == v.P)) and len(value) > 1]
+
+    # determine what the type of plot will be
+    if len(indep_comps) == 1 and len(indep_pots) == 1:
+        projection = None
+    elif len(indep_comps) == 2 and len(indep_pots) == 0:
+        projection = 'triangular'
+    else:
+        raise ValueError('The eq_plot projection is not defined and cannot be autodetected. There are {} independent compositions and {} indepedent potentials.'.format(len(indep_comps), len(indep_pots)))
     if z is not None:
         raise NotImplementedError('3D plotting is not yet implemented')
-    # TODO: Temporary workaround to fix string encoding issue when loading netcdf files from disk
-    phases = phases if phases is not None else \
-        map(str, sorted(set(np.array(eq.Phase.values.ravel(), dtype='U')) - {''}, key=str))
+    ax = plt.gca(projection=projection) if ax is None else ax
+
+    # Handle cases for different plot types
+    if projection is None:
+        x = indep_comps[0] if x is None else x
+        y = indep_pots[0] if y is None else y
+        # plot settings
+        ax.set_xlim([np.min(conds[x]) - 1e-2, np.max(conds[x]) + 1e-2])
+        ax.set_ylim([np.min(conds[y]), np.max(conds[y])])
+    elif projection == 'triangular':
+        x = indep_comps[0] if x is None else x
+        y = indep_comps[1] if y is None else y
+        # plot settings
+        ax.yaxis.label.set_rotation(60)
+        # Here we adjust the x coordinate of the ylabel.
+        # We make it reasonably comparable to the position of the xlabel from the xaxis
+        # As the figure size gets very large, the label approaches ~0.55 on the yaxis
+        # 0.55*cos(60 deg)=0.275, so that is the xcoord we are approaching.
+        # TODO: it would be nice to handle this automatically
+        ax.yaxis.label.set_va('baseline')
+        fig_x_size = ax.figure.get_size_inches()[0]
+        y_label_offset = 1 / fig_x_size
+        ax.yaxis.set_label_coords(x=(0.275 - y_label_offset), y=0.5)
+
+    # get the active phases and support loading netcdf files from disk
+    phases = map(str, sorted(set(np.array(eq.Phase.values.ravel(), dtype='U')) - {''}, key=str))
```

Comment:
There should probably be a test which verifies that `eqplot(eq)` == `eqplot(deserialize(serialize(eq))`. This could probably be done by writing the output of the LHS and RHS to files which should compare bitwise identical